crates/iota/src/client_commands.rs

```diff
@@ -3422,3 +3453,39 @@ async fn check_protocol_version_and_warn(client: &IotaClient) -> Result<(), anyh
 
     Ok(())
 }
+
+async fn select_coins_for_amount(
+    amount: u64,
+    sender: IotaAddress,
+    context: &mut WalletContext,
+) -> anyhow::Result<Vec<ObjectID>> {
+    let mut coins = Vec::new();
+
+    let mut gas_coins = context
+        .gas_objects(sender)
+        .await?
+        .iter()
+        // Ok to unwrap() since `gas_objects` guarantees gas
+        .map(|(_val, object)| GasCoin::try_from(object).unwrap())
+        .collect::<Vec<_>>();
+    // Sort in ascending order
+    gas_coins.sort_unstable_by_key(|c| c.value());
+    let mut amount_remaining = amount;
+    while amount_remaining > 0 {
+        if gas_coins.is_empty() {
+            anyhow::bail!(
+                "insufficient funds for requested amount: {amount}, available: {}",
+                amount - amount_remaining
+            );
+        }
+        let coin = if let Some(idx) = gas_coins.iter().position(|c| c.value() >= amount_remaining) {
```

Comment:
Why do we not just pop since they are sorted?

Reply:
I did it because I wanted to choose the least amount while still fulfilling the requirement, but actually I think it may not be necessary after all since its provided as the gas coin

Reply:
But the way it is at the moment, you either select the first one or you go to the other branch and pop, same thing